src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -0,0 +1,878 @@
+        self.__since = value
+
+
+# class IdBasedPaginationDetails(AbstractPaginationDetails):
+#     ...
+
+# class PageNumberBasedPaginationDetails(AbstractPaginationDetails):
+#     ...
+
+
+class AbstractCcxtInputPlugin(AbstractInputPlugin):
+
+    __DEFAULT_THREAD_COUNT: int = 3
+
+    def __init__(
+        self,
+        account_holder: str,
+        exchange_start_time: datetime,
+        native_fiat: Optional[str],
+        thread_count: Optional[int],
+    ) -> None:
+
+        super().__init__(account_holder, native_fiat)
+        self.__logger: logging.Logger = create_logger(f"{self.exchange_name()}/{self.account_holder}")
+        self.__cache_key: str = f"{str(self.exchange_name).lower()}-{account_holder}"
+        self.__client: Exchange = self.initialize_client()
+        if native_fiat:
+            self.__native_fiat: str = native_fiat
+        else:
+            self.__native_fiat = "USD"
+        if thread_count:
+            self.__thread_count: int = thread_count
+        else:
+            self.__thread_count = self.__DEFAULT_THREAD_COUNT
+        self.__markets: List[str] = []
+        self.__start_time: datetime = exchange_start_time
+        self.__start_time_ms: int = int(self.__start_time.timestamp()) * _MS_IN_SECOND
+
+    def plugin_name(self) -> str:
+        raise NotImplementedError("Abstract method")
+
+    def cache_key(self) -> Optional[str]:
+        return self.__cache_key
+
+    def initialize_client(self) -> Exchange:
+        raise NotImplementedError("Abstract method")
+
+    def exchange_name(self) -> str:
+        raise NotImplementedError("Abstract method")
+
+    def get_process_deposits_pagination_detail_set(self) -> AbstractPaginationDetailSet:
+        raise NotImplementedError("Abstract method")
+
+    def get_process_withdrawals_pagination_detail_set(self) -> AbstractPaginationDetailSet:
+        raise NotImplementedError("Abstract method")
+
+    # Some exchanges require you to loop through all markets for trades
+    def get_process_trades_pagination_detail_set(self) -> AbstractPaginationDetailSet:
+        raise NotImplementedError("Abstract method")
+
+    @property
+    def client(self) -> Exchange:
+        return self.__client
+
+    @property
+    def logger(self) -> logging.Logger:
+        return self.__logger
+
+    @property
+    def markets(self) -> List[str]:
+
+        if self.__markets:
+            return self.__markets
+
+        ccxt_markets: Any = self.__client.fetch_markets()
+        for market in ccxt_markets:
+            self.__logger.debug("Market: %s", json.dumps(market))
+            self.__markets.append(market[_ID])
+        return self.__markets
+
+    @property
+    def native_fiat(self) -> str:
+        return self.__native_fiat
+
+    @property
+    def start_time_ms(self) -> int:
+        return self.__start_time_ms
+
+    @property
+    def thread_count(self) -> int:
+        return self.__thread_count
```

Comment:
The accessor should not be duplicated here since it's already in the superclass: we can remove it from this class. But you're right about it being defined as public in `AbstractInputPlugin`. Don't worry about that: I'll take a look at it and adjust the scope.